backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -136,7 +136,7 @@ def post_collection_revision(collection_id: str, token_info: dict):
 doi_regex = re.compile(r"^10.\d{4,9}/[-._;()/:A-Z0-9]+$", flags=re.I)
```

Comment:
@Bento007 where did this come from? It appears to be a regex to match the CURIE reference portion of a DOI, but it seems different from [what I found recently and included in this PR](https://github.com/chanzuckerberg/single-cell-data-portal/blob/935de5eb499c54bc81bee3bb7b35f5edd0a76e1a/backend/corpora/lambdas/api/v1/collection.py#L227).

Reply:
Most notably it would fail to match `10.1016` which is a real doi.

Reply:
Also see https://github.com/chanzuckerberg/single-cell-data-portal/issues/3224#issuecomment-1234978730

Reply:
But I suppose this can wait to be addressed, we can take care of it when we move the Corpora API over to _only_ accepting CURIE references.

Reply:
Consolidate to a single regex. We don't want two different possible shapes of DOIs in the database.

Reply:
@ebezzi added the original regex a while back, but there is not point in having two.

Reply:
So this regex is being used by the Corpora API, and I'm explicitly not making changes to the Corpora API workflow. But since this regex is more restrictive than the one I'm using for Curation API, I think we can safely merge them without anyone having their workflow disrupted with the Corpora API 👍

Reply:
Actually that's not true. It's just different. More restrictive in some ways, less in others. Let's not update this yet.

Reply:
I would move them closer together to make it clear there are two regexs for DOI. Maybe even move them into our dedicated regex file.